### ISSUE TITLE
Shader files not being properly created

### ISSUE THREAD

**Issue by Doprez** [stride3d/stride#1799]
**Release Type**: Official Release

**Version**: 4.1.0.1898

**Platform(s)**: Windows

**Describe the bug**
Shaders require constant restarts to try and get the shaders to properly build.

**To Reproduce**
1. Create a shader
2. hope it creates the related cs file
3. restart until it may work

**Expected behavior**
It should create the cs scripts when a new SDSL file is recognized.

**Log and callstacks**
this may be realted to the VS tool extensions?
```
Error initializing Stride command proxy: System.InvalidOperationException: Could not restore Stride.VisualStudio.Commands 4.1.0.1898, this visual studio extension may fail to work properly without it. To fix this you can either build Stride.VisualStudio.Commands or pull the right version from nugget manually
   at Stride.VisualStudio.Commands.StrideCommandsProxy.<FindStrideSdkDir>d__16.MoveNext()
```

**Additional context**
https://discord.com/channels/500285081265635328/500292388820877322/1153699214807814207 a conversation on the error that promted this bug report.

**Comment by Eideren** [comment]
Might be related to #1255

**Comment by Doprez** [comment]
It definitely looks like the same result. Do you know if the Stride.VisualStudio.Commands was crashing with your report as well or if it could be related?

**Comment by Eideren** [comment]
Not sure, it's been a while. I think at the time I wasn't using the vs plugin, just looking through the gamestudio's `EffectCompilerCache` tab for errors
![image](https://github.com/stride3d/stride/assets/5742236/7a7ef720-034e-48d2-b609-771076329c7f)

**Comment by Doprez** [comment]
fixed in 4.2.0.2067
